File: src/com/mojang/escape/level/block/PitBlock.java
package com.mojang.escape.level.block;

import com.mojang.escape.Sound;
import com.mojang.escape.entities.*;
import com.mojang.escape.gui.BasicSprite;

public class PitBlock extends Block {
	private boolean filled = false;

	public PitBlock() {
		setFloorTex(1);
		setBlocksMotion(true);
	}

	public void addEntity(Entity entity) {
		super.addEntity(entity);
		if (!filled && entity instanceof BoulderEntity) {
			entity.remove();
			filled = true;
			setBlocksMotion(false);
			addSprite(new BasicSprite(0, 0, 0, 8 + 2, BoulderEntity.COLOR));
			Sound.Companion.getThud().play();
		}
	}

	public boolean blocks(Entity entity) {
		if (entity instanceof BoulderEntity) return false;
		return getBlocksMotion();
	}
}

File: common/src/commonMain/kotlin/level/block/PitBlock.kt
package com.mojang.escape.level.block

import com.mojang.escape.Sound
import com.mojang.escape.entities.BoulderEntity
import com.mojang.escape.entities.Entity
import com.mojang.escape.gui.BasicSprite

class PitBlock: Block() {
    private var filled = false

    init {
        floorTex = 1
        blocksMotion = true
    }

    override fun addEntity(entity: Entity) {
        super.addEntity(entity)
        if (!filled && entity is BoulderEntity) {
            entity.remove()
            filled = true
            blocksMotion = false
            addSprite(BasicSprite(0.0, 0.0, 0.0, 8 + 2, BoulderEntity.COLOR))
            Sound.thud.play()
        }
    }

    override fun blocks(entity: Entity): Boolean {
        if (entity is BoulderEntity) {
            return false
        }
        return blocksMotion
    }
}